### PR TITLE
Spring Boot 3.x integration. Fixes #485

### LINKED ISSUE #485: Spring Boot 3 integration
A Spring Boot 3 integration artifact is needed since the breaking changes between version 2 and 3 of Spring Boot.

### PR DESCRIPTION
**Pull request by rdebusscher**

### CODE REVIEW

File: integrations/spring-boot3/README.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+2021-08-23 15:16:02.994 DEBUG 18469 --- [           main] o.m.spring.MicrostreamConfiguration      : storage-directory : microstream_storage
+2021-08-23 15:16:02.994 DEBUG 18469 --- [           main] o.m.spring.MicrostreamConfiguration      : storage-filesystem.sql.postgres.user : postgres
+```
+
+Key values containing "password" are replaced by "xxxxx".
+
+## Build
+
+Maven build, to build just run `mvn -Pfrom_java_17 clean install`<br>
+to run integration tests run `mvn -Prun-its -Pfrom_java_17 clean install`. Integration tests require Docker to run.
```
Comment (zdenek-jonas) [inline integrations/spring-boot3/README.md]: The profile  `-Pfrom_java_17` is active one level higher. For the building project as such, it has no effect. I would remove it. And Just write, that this build need at least java 17.

File: integrations/spring-boot3/src/main/java/one/microstream/integrations/spring/boot/types/mongodb/Mongodb.java [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+ * 
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License, v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is
+ * available at https://www.gnu.org/software/classpath/license.html.
+ * 
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ * #L%
+ */
```
Comment (zdenek-jonas) [inline integrations/spring-boot3/src/main/java/one/microstream/integrations/spring/boot/types/mongodb/Mongodb.java]: The license in this file is 2x and the second time with a different artifact id.

**Comment by zdenek-jonas** [comment]
Build with integration tests failed
`[ERROR] Failed to execute goal org.apache.maven.plugins:maven-invoker-plugin:3.0.1:integration-test (runOneTestToSetupRepoAndAvoidConcurrentDownloads) on project microstream-integrations-spring-boot3: Failed to interpolate file /Users/zdenekjonas/devel/microstream/integrations/spring-boot3/src/it/settings.xml: /Users/zdenekjonas/devel/microstream/integrations/spring-boot3/src/it/settings.xml (No such file or directory) -> [Help 1]`

Missing settings.xml in the root of integration tests projects.

**Comment by zdenek-jonas** [comment]
Please consider moving the same classes to a library such as microstream-spring-configuration-commons or similar. To avoid code duplication.

**Comment by zdenek-jonas** [comment]
Also add in file:
maven_run_its.yml
section for run ITS build under Java17.  It is sufficient to run these tests in this module only.

Please check the PR https://github.com/microstream-one/microstream/pull/491 first

**Comment by rdebusscher** [comment]
> consider moving the same classes to a library such as microstream-spring-configuration-commons or similar. 

Will do after I have finalised https://github.com/microstream-one/microstream/pull/490 as that PR makes such a refactor now  difficult.

**Comment by rdebusscher** [comment]
On hold until #490 is done.

**Comment by rdebusscher** [comment]
Creation of a common artifact in a way

- It is compatible with current version 7.1 (no changes needed in user code, nor pom.xml)
- No code changes required when moving from Spring Boot 2 to 3 version (just pom.xml to update)

**Comment by rdebusscher** [comment]
Unrecoverable Git merge mess. Redo

**Pull request closed without merging**